src/Http/Controllers/VoyagerController.php

```diff
@@ -68,8 +68,54 @@ public function upload(Request $request)
         return "<script> parent.helpers.setImageValue('".Voyager::image($fullFilename)."'); </script>";
     }
 
-    public function profile()
+    public function profile(Request $request)
     {
-        return Voyager::view('voyager::profile');
+        $id = auth()->user()->id;
+
+        return redirect()->route('voyager.users.showProfile', $id);
+    }
+
+    public function showProfile(Request $request, $id)
+    {
+        if ($id == auth()->user()->id) {
+            return Voyager::view('voyager::profile');
+        } else {
+            abort(404);
+        }
+    }
+
+    public function editProfile(Request $request, $id)
+    {
+        if ($id == auth()->user()->id) {
+            $voyagerBaseController = new VoyagerBaseController();
+
+            return $voyagerBaseController->edit($request, $id);
+        } else {
+            abort(404);
+        }
+    }
+
+    public function updateProfile(Request $request, $id)
+    {
+        if ($id == auth()->user()->id) {
+            // If can't edit users can't change his own role but need to preserve roles set
+            if (!auth::user()->hasPermission('edit_users')) {
```

Comment:
I'd rather see permissions checks use the Laravel Authorization functionality than direct permissions checks.  Keeps it consistent